### DISCUSSION TITLE
Need guidance regarding development of Solid beam formulation.

### DISCUSSION THREAD

**Discussion by abdullah2146** [idaholab/moose#21229] [Q&A Modules: Solid mechanics] [answered]
I want to develop the solid beam formulation on MOOSE as it's presented in this journal paper https://www.sciencedirect.com/science/article/pii/S0045782513001618
Is there any example or C++ files that would be helpful in developing this?
The problem I am facing is how I will define the entities of B matrix in MOOSE which need to be calculated at the nodes of the plane. 
![image](https://user-images.githubusercontent.com/104014684/172486452-c1785aa5-12f9-4c45-907e-59797237779e.png)
  
![image](https://user-images.githubusercontent.com/104014684/172486517-9f6d94c4-70a7-4435-b970-d7b4cf58d55a.png)

@jain651 may be you can help me, as I asked this thing in MOOSE workshop and the presenter of Tensor mechanics module told me to ask you.

**Comment by jain651** [comment, marked as answer]
Please have a look at the formulation of the shell elements. Their formulation also involves formation of matrices like these. I hope this will help you.
